### PR TITLE
Building a Data Practice from Scratch

### PR DESCRIPTION
**Pull request by selftext**
Here's my post with advice for new data leads at early stage startups.

Some of this feels obvious to me (and I'd love feedback if that's the case), but I've also seen a number of organizations that didn't invest in these areas early on, and it was clear they were suffering because of it.

I would also appreciate feedback on what could be cut. As much as I'm laconic in speech, I'm logorrheic in writing. Is anything too wordy or redundant?

On the flip side, it would be great to know if anything could be better fleshed out with specific examples.

Thanks!

### CODE REVIEW

File: content/post/building-a-data-practice.md [hunk 1]
```diff
@@ -0,0 +1,85 @@
+---
+title: "Building a Data Practice from Scratch"
+author: "Jeremy Salfen"
+cover: "/img/cover.jpg"
+tags: ["process", "communication", "documentation", "project management"]
+date: 2018-10-29
+draft: false
+---
+
+The first data hires at an early stage startup face numerous challenges — an infrastructure focused more on running the business than analyzing it, an organization hungry for information without a process for requesting and prioritizing it, and little documentation on how anything is done. What should they do first?
```
Comment (mikekaminsky) [inline content/post/building-a-data-practice.md]: This phrase scans strangely for me

> an infrastructure focused more

Maybe

> infrastructure build to run the business but not analyze it

File: content/post/building-a-data-practice.md [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+
+<!--more-->
+
+I recently found myself in a position like this at Sawyer, a platform for children's educational activities. There are good resources for startup founders hiring for analytics — I highly recommend [Tristan Handy's post](https://thinkgrowth.org/the-startup-founders-guide-to-analytics-1d2176f20ac1) — but not much has been written for data leads at a new organization.
```
Comment (mikekaminsky) [inline content/post/building-a-data-practice.md]: `[Sawyer](https://www.hisawyer.com/)`

File: content/post/building-a-data-practice.md [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+
+In this post I share some ideas on tools and processes that have been helpful for me. Of course, you should also be thinking about business analysis — defining key metrics, generating financial reports, and doing strategic research. You will also have to think about data infrastructure — databases, pipelines, ETL, BI tools. Those are big topics that are better left for other posts. Here I focus on building a data practice, a set of tools and processes that will be the foundation for all your analysis and infrastructure work.
+
+There are a few principles woven into the suggestions below:
+
+* Don't worry about making things fancy. Do the simplest thing now that works.
```
Comment (mikekaminsky) [inline content/post/building-a-data-practice.md]: > simplest thing now that works

to

> simplest thing that works now

File: content/post/building-a-data-practice.md [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+* Keep an eye on how things will scale, but rein in your impulses to optimize them.
```
Comment (mikekaminsky) [inline content/post/building-a-data-practice.md]: Maybe add "Analytics should be lean and agile, too!"

File: content/post/building-a-data-practice.md [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+
+### Create a data dictionary
+As you learn things, especially things that are surprising, write them down. Think about what the next data hire will need to know. Avoid the urge to make this fancy — a Google doc is fine.
+
+The concept of a data dictionary is a bit controversial. What do you include? What don't you include? How do you maintain it? For more mature organizations, I recommend considering [Carl Anderson's suggestions](https://medium.com/@leapingllamas/data-dictionary-a-how-to-and-best-practices-a09a685dcd61) for using a data dictionary to create a shared understanding around key metrics and vocabulary.
+
+In this case, though, a data dictionary is primarily for you and future data hires. Even though the business and its data will change over time, there are still key pieces of information that are useful to document. Are timezones in UTC or your organization's time? What are some core pieces of business logic? How are key metrics calculated? Are there important database tables or column names that are not self-explanatory?
+
+At Kickstarter, we called this document *Common Data Gotchas*, which is a good way to think about it.
```
Comment (mikekaminsky) [inline content/post/building-a-data-practice.md]: I really like this -- I think it's key to _not_ get hung up on making it exhaustive (a list of every column should be an explicit anti-goal). You could maybe drive that home with something like:

> Do _not_ set yourself the goal of creating an exhaustive data dictionary. The completionists find it challenging to leave all of those columns undocumented, but not only will it be time consuming to create, a pain to manage, and seldom read, but despite your best efforts it will get quickly out of date.

File: content/post/building-a-data-practice.md [hunk 6]
```diff
@@ -0,0 +1,85 @@
...
+
+### Use a project management tool
+Start tracking all the work you do, even quick data requests. Write down the question, the results, and how you got them. Encourage others in your organization to interact with you in this tool. Documenting your work in this way will be important for reproducibility, it will be good process as your data team grows, and it will help with visibility and transparency, demonstrating your impact on the business.
```
Comment (mikekaminsky) [inline content/post/building-a-data-practice.md]: I think you could flesh the "why" out a bit more here.

File: content/post/building-a-data-practice.md (hunk 5 above)
Comment (ilanman) [reply content/post/building-a-data-practice.md]: Agreed. I would add that even before describing columns, you can just get alignment and clarity around definitions of key concepts. For example, when we (at Trialspark) talk about "active studies", different teams have different use cases and definitions. It's helpful to align on the concept first ("active recruitment studies" "active FDA studies" etc...) and then figure out where and how that maps to the columns.

File: content/post/building-a-data-practice.md (hunk 6 above)
Comment (ilanman) [reply content/post/building-a-data-practice.md]: Yes - and maybe something like "be careful not to make this process burdensome for stakeholders too early on"

File: content/post/building-a-data-practice.md [hunk 7]
```diff
@@ -0,0 +1,85 @@
...
+
+My first choice for this is Trello, but use whatever integrates well into your organization.
+
+### Create communication channels for data
```
Comment (ilanman) [inline content/post/building-a-data-practice.md]: Another consideration:

These few paragraphs are related to the topic of setting stakeholder expectations. I've found that there are two classes of coworkers (here anyway):

1. Those who've worked with Data teams and have a certain expectation/vision.
2. Those who haven't and aren't sure what to expect.

You'll have to temper the expectations of the former, e.g. "Looker won't be up and running tomorrow and here are the reasons why". I think the expectations of the latter group are well addressed in this post.

File: content/post/building-a-data-practice.md [hunk 8]
```diff
@@ -0,0 +1,85 @@
...
+There are a few principles woven into the suggestions below:
+
+* Don't worry about making things fancy. Do the simplest thing now that works.
+* Keep an eye on how things will scale, but rein in your impulses to optimize them.
+* Documentation, transparency, and reproducibility are interrelated and fundamental. Start good habits in these areas now, but expect to iterate on them and change how things are done as the organization matures and grows.
+
+With those principles in mind, here are my recommendations for what to do in your first weeks at an early stage startup.
+
+### Map the data infrastructure
+Start documenting all the places where data is stored, processed, and accessed. Does your organization have multiple databases? Does it have external data sources? Are there systems moving data from one place to another? How are people interacting with data? Write this all down, and include links and information about how to get access.
```
[the remark this answers is not in the document]
Comment (selftext) [reply content/post/building-a-data-practice.md]: Great point. I don't think it's necessary to go deep on everything — e.g. I made a note that our salespeople use Dialpad for calls, but that was it — but I do think it's good to document everything you find. You never know what will end up being important later.

I'll add a few sentences about this.

File: content/post/building-a-data-practice.md (hunk 7 above)
Comment (selftext) [reply content/post/building-a-data-practice.md]: Another great point. You can definitely tell the kinds of groups I've interacted with the most. :)

I think going deeper on stakeholder management is a little outside the scope here, but that could make an interesting post in the future.
